File: rust/src/utils/config_util.rs
use anyhow::{Result, bail};
use std::env;

/// Gets a required environment variable or returns an error if missing/empty.
///
/// # Arguments
/// * `var` - name of the environment variable
///
/// # Returns
/// * `String` value of the environment variable
pub fn get_required_env(var: &str) -> Result<String> {
    match env::var(var) {
        Ok(val) if !val.trim().is_empty() => Ok(val),
        Ok(_) => bail!("Environment variable {var} is set but empty"),
        Err(_) => bail!("Environment variable {var} is missing"),
    }
}

File: node-express/src/utils/configUtil.ts
/**
 * @author DiZed Team
 * @copyright Copyright (c) DiZed Team (https://github.com/di-zed/)
 */

/**
 * Config Utility.
 */
class ConfigUtil {
  /**
   * Gets a required environment variable or returns an error if missing/empty.
   *
   * @param name
   * @returns string
   */
  public getRequiredEnv(name: string): string {
    const value = process.env[name];
    if (value === undefined || value === null || value.trim() === '') {
      throw new Error(`Environment variable ${name} is required but not set`);
    }

    return value;
  }
}

export default new ConfigUtil();
